test/integration/symlink-git-dir.test.js

```diff
@@ -14,9 +15,10 @@ describe('lint-staged', () => {
   test(
     'supports symlinked git dir',
     withGitIntegration(async ({ cwd, execGit, gitCommit, readFile, writeFile }) => {
-      // Rename `.git` to `git` and add symbolic link pointing to it
-      await fs.rename(path.resolve(cwd, '.git'), path.resolve(cwd, 'git'))
-      await fs.symlink(path.resolve(cwd, 'git'), path.resolve(cwd, '.git'), 'dir')
+      const tempDir = await createTempDir()
+      // Move `.git` to tempDir and add symbolic link pointing to it
+      await fs.rename(path.resolve(cwd, '.git'), path.resolve(cwd, tempDir))
+      await fs.symlink(path.resolve(cwd, tempDir), path.resolve(cwd, '.git'), 'dir')
```

Comment:
This test failed before I set `--hide-untracked` to `false` by default, so I had to move the `git/` directory somewhere else.  Is it expected that a user might symlink `.git/` to something in the same directory?

Reply:
Yes, this expected functionality that _lint-staged_ should handle, hence the test case.

Reply:
But then it would certainly be added to the `.gitignore`, right?  Otherwise the hidden git files would be visible to git.

Reply:
Or it would be named `**/.git/`, and therefore invisible to git.

Reply:
I understand that support for symlinked `.git/` is necessary, but not why the link target would ever be allowed to show up as untracked files in the same repository.

Reply:
IMO this would be a user error where the itself file is added first, and only after that `.gitignore` is updated.

It's good to have both of these flags disabled by default to make sure _lint-staged_ by default doesn't touch any non-relevant files (staged) unless the user really needs it.

Sorry, maybe I'm not understanding something here.